container/container.go

```diff
@@ -949,3 +949,26 @@ func (container *Container) InitializeStdio(iop libcontainerd.IOPipe) error {
 
 	return nil
 }
+
+// SecretMountPath returns the path of the secret mount for the container
+func (container *Container) SecretMountPath() string {
+	return filepath.Join(container.Root, "secrets")
+}
+
+// SecretFilePath returns the path to the location of a secret on the host.
+func (container *Container) SecretFilePath(secretRefTarget *swarmtypes.SecretReferenceFileTarget) (string, error) {
+	secretMountPath := container.SecretMountPath()
+	mountPath := filepath.Join(secretMountPath, secretRefTarget.Name)
```

Comment:
It does allow path separators - that's the point of https://github.com/docker/swarmkit/pull/2118.

The check below tries to ensure that this doesn't escape the mount point, but I'd appreciate careful review to make sure this works as intended.